helpers/helpers.v2.1.d/backup

```diff
@@ -0,0 +1,318 @@
+#!/bin/bash
+
+CAN_BIND=${CAN_BIND:-1}
+
+# Add a file or a directory to the list of paths to backup
+#
+# usage: ynh_backup /path/to/stuff
+#
+# NB : note that this helper does *NOT* perform any copy in itself, it only
+# declares stuff to be backuped via a CSV which is later picked up by the core
+#
+# NB 2 : there is a specific behavior for $data_dir (or childs of $data_dir) and
+# /var/log/$app which are *NOT* backedup during safety-backup-before-upgrade,
+# OR if the setting "do_not_backup_data" is equals 1 for that app
+#
+# The rationale is that these directories are usually too heavy to be integrated in every backup
+# (think for example about Nextcloud with quite a lot of data, or an app with a lot of media files...)
+#
+# This is coupled to the fact that $data_dir and the log dir won't be (and
+# should NOT) be deleted during remove, unless --purge is used. Hence, if the
+# upgrade fails and the script is removed prior to restoring the backup, the
+# data/logs are not destroyed.
+#
+ynh_backup() {
+
+    local target="$1"
+    local is_data=false
+
+    # If the path starts with /var/log/$app or $data_dir
+    if ([[ -n "${app:-}" ]] && [[ "$target" == "/var/log/$app*" ]]) || ([[ -n "${data_dir:-}" ]] && [[ "$target" == "$data_dir*" ]])
```

Comment:
> --is_big behavior is replaced by checking if the path is $data_dir (or a child) or /var/log/$app

Yeah that might be a bit restrictive and what if `$data_dir/some_important_config_file.json` ?

Reply:
Also packagingv3™ but those directories could be declared in manifest.

Reply:
>Yeah that might be a bit restrictive and what if $data_dir/some_important_config_file.json ?

yeah idk, i tried to study carefully what was really done in the wild ... it's still unclear what policy we want exactly (for example it wasnt clear for log files when we discussed it on the chat the other day)

dunno what do about conf inside $data_dir ... but the main point is that the specific behavior basically happens only for safety-backups-before-upgrade (..or if people enabled backup_core_only manually...) hence the data_dir will be neither backuped nor removed (cf the --purge mechanism) -- and we REALLY have to get the two side consistent (the backup side and remove (prior to restore ofc). Currently there may be apps which do not declare their data_dir as "is_big" (sigh) hence might get simply rmed with no backup if an upgrade fails...